core/src/Kokkos_Concepts.hpp

```diff
@@ -326,33 +326,44 @@ struct is_space {
   // For backward compatibility, deprecated in favor of
   // Kokkos::Impl::HostMirror<S>::host_mirror_space
 
-  using host_memory_space = typename std::conditional<
+ private:
```

Comment:
Change `impl_` to something that clearly indicates not to use them within Kokkos.
Comment that you define these aliases to avoid deprecated warnings below in the definition of `host_mirror_space`